samples/apps/copilot-chat-app/webapi/Auth/ApiKeyAuthenticationHandler .cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Options;
+
+namespace SemanticKernel.Service.Auth;
+
+/// <summary>
+/// Class implementing API key authentication.
+/// </summary>
+public class ApiKeyAuthenticationHandler : AuthenticationHandler<ApiKeyAuthenticationSchemeOptions>
+{
+    public const string AuthenticationScheme = "ApiKey";
+    public const string ApiKeyHeaderName = "x-api-key";
+
+    private readonly IOptionsMonitor<ApiKeyAuthenticationSchemeOptions> _options;
+
+    /// <summary>
+    /// Constructor
+    /// </summary>
+    public ApiKeyAuthenticationHandler(
+        IOptionsMonitor<ApiKeyAuthenticationSchemeOptions> options,
+        ILoggerFactory logger,
+        UrlEncoder encoder,
+        ISystemClock clock) : base(options, logger, encoder, clock)
+    {
+        this._options = options;
+    }
+
+    protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        if (!this.Request.Headers.ContainsKey(ApiKeyHeaderName))
+        {
+            return Task.FromResult(AuthenticateResult.Fail("No API key provided"));
+        }
+
+        if (string.IsNullOrWhiteSpace(this.Options.ApiKey))
```

Comment:
[nit] Verify the service configuration is valid before checking if the request contains the header.